src/PRJobProducer/Program.cs

```diff
@@ -0,0 +1,359 @@
+            {
+                if (pr.UpdatedAt > CommentCutoffDate)
+                {
+                    var comments = await client.Issue.Comment.GetAllForIssue(Owner, Repo, pr.Number);
+
+                    for (var i = comments.Count - 1; i >= 0; i--)
+                    {
+                        var comment = comments[i];
+
+                        if (comment.CreatedAt < CommentCutoffDate)
+                        {
+                            break;
+                        }
+
+                        if (comment.Body.StartsWith(BenchmarkRequest) && await client.Organization.Member.CheckMember(Owner, comment.User.Login))
+                        {
+                            yield return pr;
+                        }
+                        else if (comment.User.Login.Equals(botLoginName, StringComparison.OrdinalIgnoreCase))
+                        {
+                            // The bot has already commented with results for the most recent benchmark request.
+                            break;
+                        }
+                    }
+                }
+            }
+        }
+
+        private static async Task<string[]> GetBuildCommands()
+        {
+            using var processingJsonStream = File.OpenRead(BaseJobPath);
+            using var jsonDocument = await JsonDocument.ParseAsync(processingJsonStream);
+            BuildInstructions buildInstructions = null;
+
+            foreach (var element in jsonDocument.RootElement.EnumerateObject())
+            {
+                if (element.NameEquals("BuildInstructions"))
+                {
+                    buildInstructions = JsonSerializer.Deserialize<BuildInstructions>(element.Value.GetRawText());
+                    break;
+                }
+            }
+
+            if (buildInstructions is null)
+            {
+                throw new InvalidDataException("Job file doesn't include a valid 'BuildInstructions' property");
+            }
+
+            return buildInstructions.BuildCommands;
+        }
+
+        private static async Task RequestBenchmark(PullRequest pr, string session)
+        {
+            var baseFile = new FileInfo(BaseJobPath);
+            var newJobPath = Path.Combine(Path.GetTempPath(), $"{session}.{baseFile.Name}");
+            var newJobFile = new FileInfo(newJobPath);
+
+            baseFile.CopyTo(newJobPath);
+
+            try
+            {
+                using (var newJobStream = File.Open(newJobPath, System.IO.FileMode.Open))
+                {
+                    var jsonDictionary = await JsonSerializer.DeserializeAsync<Dictionary<string, object>>(newJobStream);
+
+                    jsonDictionary["BuildInstructions"] = new BuildInstructions
+                    {
+                        BuildCommands = BuildCommands,
+                        BaselineSHA = pr.Base.Sha,
+                        PullRequestSHA = pr.Head.Sha,
+                    };
+
+                    newJobStream.SetLength(0);
```

Comment:
Is this needed?

Reply:
If the new contents of the file are shorter than the old contents, yes. Before adding whitespace to the serialized output, I would see the truncated end of the original file after the serialized output if I didn't call SetLength(0). Because I cannot guarantee that the base job file doesn't have a ton of extra whitespace or something that could make it larger than the new job file, I kept it.

Reply:
Add a comment? I wouldn't have guessed that.